File: ProyectoCivitasJava/src/civitas/Casilla.java
/**
 * Proyecto Civitas
 * Sergio Hervás Cobo
 * Javier Suárez Lloréns
 * Programación y Diseño Orientado a Objetos
 * 2º B3 - Grado en Ingeniería Informática
 * E.T.S.I. Informática y de Telecomunicación
 * Universidad de Granada
 */

package civitas;
import java.util.ArrayList;

public class Casilla {
    
    //Atributos de instancia
    private String nombre;  
    
    Casilla(String nombre){ //constructor de la clase    
        this.nombre = nombre;
    }
    
    public String getNombre(){
        return nombre;
    }
    
    protected void informe(int iactual, ArrayList <Jugador> todos){
        Diario.getInstance().ocurreEvento("El jugador " + todos.get(iactual).getNombre() + " ha caído en la Casilla: " + toString());
    }
    
    public boolean jugadorCorrecto(int iactual, ArrayList <Jugador> todos){
        boolean comprobacion = false;
        if(iactual >= 0 && iactual < todos.size()){
            comprobacion = true;
        }
        return comprobacion;
    }
    
    void recibeJugador(int iactual, ArrayList <Jugador> todos){
        if(jugadorCorrecto(iactual, todos))
            informe(iactual, todos);
    }
    
    @Override
    public String toString(){
        String salida = nombre;

        return salida;
    }
}

File: ProyectoCivitasRuby/lib/modelo/casilla.rb


require_relative "titulo_propiedad"
require_relative "sorpresa"
require_relative "jugador"
require_relative "diario"
require_relative "mazo_sorpresas"

module Civitas  
  class Casilla
    
    #Constructores
    def initialize(nombre)
      @nombre = nombre
    end
  
    
    #MÉTODOS PÚBLICOS
    public    
    
    #Lectores
    attr_reader :nombre
   

    def jugador_correcto(actual, todos)
      (actual < todos.size && actual >= 0)
    end
    
    
    def recibe_jugador(iactual, todos)
        informe(iactual, todos)
    end
        
        
    def to_s
      salida = "Nombre: #{@nombre} Tipo: #{@tipo}"
      salida
    end
    
    
    #MÉTODOS PROTEGIDOS
    protected
       
    def informe(actual, todos)
      Diario.instance.ocurre_evento("El jugador " + "#{todos[actual].nombre} ha caido en la casilla #{@nombre}" + "\n\n")
    end
   
  end
end